tests/Unit/CrudPanel/CrudPanelAccessTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Backpack\CRUD\Tests\Unit\CrudPanel;
+
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class CrudPanelAccessTest extends BaseCrudPanelTest
+{
+    private $unknownPermission = 'unknownPermission';
+
+    private $defaultAccessList = [
+        'list',
+        'create',
+        'update',
+        'delete',
+    ];
+
+    private $fullAccessList = [
+        'list',
+        'create',
+        'update',
+        'delete',
+        'revisions',
+        'reorder',
+        'show',
+        'details_row',
+    ];
+
+    public function testDefaultAccess()
+    {
+        $this->assertEquals($this->defaultAccessList, $this->crudPanel->access);
+    }
+
+    public function testHasAccess()
+    {
+        foreach ($this->defaultAccessList as $permission) {
+            $this->assertTrue($this->crudPanel->hasAccess($permission));
+        }
+
+        foreach (array_diff($this->fullAccessList, $this->defaultAccessList) as $permission) {
+            $this->assertFalse($this->crudPanel->hasAccess($permission));
+        }
+    }
+
+    public function testAllowAccess()
+    {
+        $permission = 'reorder';
+
+        $this->crudPanel->allowAccess($permission);
+
+        $this->assertTrue($this->crudPanel->hasAccess($permission));
+    }
+
+    public function testAllowAccessToUnknownPermission()
+    {
+        $this->markTestIncomplete();
+
+        // TODO: do we allow any value for permissions?
+        $this->crudPanel->allowAccess($this->unknownPermission);
```

Comment:
@tabacitu, was this done to allow anyone to add custom permissions?

Reply:
Yup! Exactly.